### PR TITLE
[WIP] Windows support

### PR DESCRIPTION
**Pull request by isuruf**
Depends on https://github.com/conda-forge/numpy-feedstock/pull/73

### CODE REVIEW

**Comment by isuruf** [comment]
@xoviat, do you know the reason for the failure?

**Comment by xoviat** [comment]
Are you 100% sure that numpy is linked against openblas and libflang? Where was that numpy built?

**Comment by isuruf** [comment]
There are 2 problems
- Default appveyor image has VS 2015 Update 3 and cl.exe crashes with an ICE
- VS 2015 Update 2 can't import numpy. I tried compiling and testing numpy on that image, but it can't import numpy.

@xoviat, do you know if we can compile scipy with clang-cl.exe?

**Comment by xoviat** [comment]
Theoretically possible but the time commitment is a open ended. The question is why can I compile scipy but you can't.

**Comment by xoviat** [comment]
See here: https://ci.appveyor.com/project/xoviat/scipy/build/1.0.53

**Comment by isuruf** [comment]
This is the error,
```
c:\bld\scipy_1512059958270\work\scipy-1.0.0\scipy\stats\_stats.c(15362) : fatal error C1001: An internal error has occurred in the compiler.
(compiler file 'f:\dd\vctools\compiler\utc\src\p2\main.c', line 255)
 To work around this problem, try simplifying or changing the program near the locations listed above.
Please choose the Technical Support command on the Visual C++ 
 Help menu, or open the Technical Support help file for more information
INTERNAL COMPILER ERROR in 'C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\link.exe'
    Please choose the Technical Support command on the Visual C++
    Help menu, or open the Technical Support help file for more information
```
Internal compiler errors are hard to reproduce.

**Comment by xoviat** [comment]
Idea: compile numpy with vs2017 and numpy PR #9965.

**Comment by xoviat** [comment]
Then compile scipy with vs2017.

**Comment by isuruf** [comment]
Can `numpy.distutils` use VS2017?

**Comment by xoviat** [comment]
Yes

**Comment by isuruf** [comment]
How do I use VS2017?

**Comment by xoviat** [comment]
Put os: Visual Studio 2017 in appveyor.yml

**Comment by isuruf** [comment]
Yes, but that doesn't mean numpy.distutils will pick up VS2017

**Comment by xoviat** [comment]
On the latest Python 3.6 it will. On older versions it still will if vs 2015 is not found.

**Comment by isuruf** [comment]
Fixed in the last commit. See the commit message. Your logs helped a lot. Thanks.

**Comment by isuruf** [comment]
26/12056 failed for flang/openblas which is more than flang/mkl. I think OpenBLAS is not well tested in windows with MSVC. I'll look into running OpenBLAS test-suite.

**Comment by isuruf** [comment]
@conda-forge-admin, please rerender

**Comment by jschueller** [comment]
Hi @isuruf what's up on this ? Are there any important defects revealed by the testsuite here of could we ignore some ?

**Comment by jakirkham** [comment]
Looking at the failures, they are things like computing decompositions, determinants, inversions, solving for matrices, etc. These are things that I would typically associate with LAPACK. Though could be wrong. Did not check the code of the failing tests.

Assuming LAPACK is the culprit, seeing failures on Windows here isn't too surprising given the LAPACK part of the OpenBLAS code base is basically unused on Windows. Perhaps we should look into alternative LAPACK implementations? OpenBLAS 0.20.0 includes ReLAPACK support, which could be a nice option. Raised issue ( https://github.com/conda-forge/openblas-feedstock/issues/43 ) to explore it. Another option would be to use our own build of Netlib's LAPACK. Other suggestions?

**Comment by tadeu** [comment]
Thanks for following up on this @jakirkham, getting SciPy to work on Windows is really not an easy task :)

I'll add some things I just dig hoping that maybe you find something useful there:

SciPy has an official build "script" used for testing on AppVeyor here: https://github.com/scipy/scipy/blob/master/appveyor.yml

They use OpenBLAS and `mingw-gfortran`, and it seems they use the LAPACK that is provided with OpenBLAS [side note: weird that a BLAS lib provides a LAPACK implementation and not the other way around].  There is this [documentation](http://scipy.github.io/devdocs/building/windows.html) about building SciPy on Windows that also seems to confirm it.

Their builds are currently passing here: https://ci.appveyor.com/project/scipy/scipy

**Comment by tadeu** [comment]
Hmm, this failure indicate that it might actually be a problem with LAPACK configuration for complex numbers?

```
_________________________ TestInv.test_random_complex _________________________
self = <scipy.linalg.tests.test_basic.TestInv object at 0x000000B45F8A9400>
    def test_random_complex(self):
        n = 20
        for i in range(4):
            a = random([n, n])+2j*random([n, n])
            for i in range(n):
                a[i, i] = 20*(.1+a[i, i])
            a_inv = inv(a)
            assert_array_almost_equal(dot(a, a_inv),
>                                     identity(n))
E           AssertionError: 
E           Arrays are not almost equal to 6 decimals
E           
E           (mismatch 80.0%)
E            x: array([[ 9.787941e-01-1.260238e-02j, -1.318547e-02-1.933605e-03j,
E                   -1.024285e-01-5.050906e-04j, -1.695463e-02+2.560524e-04j,
E                   -1.235516e-02-1.183319e-02j, -1.123503e-02-2.985081e-03j,...
E            y: array([[1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
E                   0., 0., 0., 0.],
E                  [0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,...
a          = array([[5.83038901e+00+2.85997553e+01j, 6.22108771e-01+1.44818297e+00j,
        4.37727739e-01+3.73528788e-02j, 7.8535...3467485e-01j, 9.02555026e-01+7.41271734e-01j,
        2.25995526e-01+1.51916906e+00j, 8.08307477e+00+1.22473564e+01j]])
a_inv      = array([[ 6.72549153e-03-3.38476215e-02j, -3.15675610e-04+1.29006546e-03j,
         1.78181454e-03+6.08326700e-04j,  4....0989e-04j, -4.39107047e-04+1.62316713e-03j,
        -2.65091529e-03+1.90687418e-03j,  3.81396360e-02-5.72156374e-02j]])
i          = 19
n          = 20
self       = <scipy.linalg.tests.test_basic.TestInv object at 0x000000B45F8A9400>
```

**Comment by jakirkham** [comment]
> I have no idea why it finds it twice in conda-forge, and why the language is `f77` in one an `c` in the other (perhaps it's because setup couldn't find an `msvc` compiler in the scipy official build).

It might be picking up LAPACKE, which are the C bindings to LAPACK. Just guessing though.

**Comment by FSund** [comment]
I'm really keen to get this working, but I have no idea how I can help.

How do I get started? I have never worked on a feedstock before.
How do I compile the feedstock on my own machine?
Are the only files that need to be changed `conda-forge.yml` and the files in the `recipe` folder?

Sorry if this is not the correct place for these questions.

**Comment by tadeu** [comment]
@FSund try to start from here: https://conda-forge.org/docs/buildwin.html#local-testing 

You should be able to build it locally by cloning this recipe and using `conda-build`: https://conda.io/docs/commands/build/conda-build.html

There's a chat here where you might find help: https://gitter.im/conda-forge/conda-forge.github.io

**Comment by nicoddemus** [comment]
It seems scipy is the deprecated MarkInfo objects. I'm afraid this needs to be addressed upstream. Meanwhile I suggest to pin `pytest` to `<4`.

**Comment by FSund** [comment]
Thanks. Perhaps `pytest <4` should be added to the recipe? I added it to `requirements: run:` and `test: requires:` on my copy. That did the trick for me, but not sure where it should really go.

I'm running into a new issue with one of the tests. Specifically this one, which makes python crash

    linalg/tests/test_decomp_update.py::TestQRinsert_D::test_economic_1_col_bad_update 

More detailed logs:

    ===== testing package: scipy-1.0.1-py36_blas_openblas_200 =====
    running run_test.py
    ============================= test session starts =============================
    platform win32 -- Python 3.6.6, pytest-3.10.1, py-1.7.0, pluggy-0.8.0 -- C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542705286025\_test_env\python.exe

    [...]

    linalg/tests/test_decomp_update.py::TestQRinsert_D::test_economic_1_row PASSED [ 14%]
    linalg/tests/test_decomp_update.py::TestQRinsert_D::test_economic_p_row PASSED [ 14%]
    linalg/tests/test_decomp_update.py::TestQRinsert_D::test_economic_1_col PASSED [ 14%]
    linalg/tests/test_decomp_update.py::TestQRinsert_D::test_economic_1_col_bad_update 
    (C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542705286025\_test_env) C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542705286025\test_tmp>IF 255 NEQ 0 exit 1 
    Tests failed for scipy-1.0.1-py36_blas_openblas_200.tar.bz2 - moving package to C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\broken
    WARNING:conda_build.build:Tests failed for scipy-1.0.1-py36_blas_openblas_200.tar.bz2 - moving package to C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\broken
    TESTS FAILED: scipy-1.0.1-py36_blas_openblas_200.tar.bz2

Seems like this could be the same issue: https://github.com/scipy/scipy/issues/5338
I tried adding [this](https://github.com/scipy/scipy/pull/8214) to patches, but get the same result. I'm not sure if I added the patch correctly though.

Here is my `fix_error.patch`

    diff --git a/scipy/linalg/_decomp_update.pyx.in b/scipy/linalg/_decomp_update.pyx.in
    index 03875ff7d92..dbaa9061cdc 100644
    --- a/scipy/linalg/_decomp_update.pyx.in
    +++ b/scipy/linalg/_decomp_update.pyx.in
    @@ -1890,6 +1890,8 @@ cdef qr_insert_col(Q, R, u, int k, rcond, bint overwrite_qru, bint check_finite)
         cdef bint economic
         cdef float frc = libc.float.FLT_EPSILON
         cdef double drc = libc.float.DBL_EPSILON
    +    cdef float_complex cfrc = frc
    +    cdef double_complex cdrc = drc
     
         # 1 eco q alloc, r alloc, u any
         # p eco q alloc, r alloc, u any unless eco->fat then F
    @@ -1963,7 +1965,7 @@ cdef qr_insert_col(Q, R, u, int k, rcond, bint overwrite_qru, bint check_finite)
             rptr = extract(rnew, rs)
             uptr = extract(u1, us)
     {{py:
    -RCONDS = ['&frc', '&drc', '<float_complex*>&frc', '<double_complex*>&drc']
    +RCONDS = ['&frc', '&drc', '<float_complex*>&cfrc', '<double_complex*>&cdrc']
     }}
             with nogil:
                 {{for COND, TYPECODE, CNAME, RC in zip(CONDS, TCODES, CNAMES, RCONDS)}}

which I added in `meta.yaml`

    source:
      url: https://pypi.io/packages/source/s/scipy/scipy-{{ version }}.tar.gz
      sha256: 8739c67842ed9a1c34c62d6cca6301d0ade40d50ef14ba292bd331f0d6c940ba
      patches:
        - delete_test_segfault.patch
        - fix_error.patch

It gets applied right before the resource usage statistics is listed, so I think I did it correctly, but the patch itself might not be correctly formatted.

    Applying patch: 'C:\\Users\\IEUser\\code\\scipy-feedstock\\recipe\\fix_error.patch'
    source tree in: C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542711051699\work

    Resource usage statistics from building scipy:
       Process count: 5
       CPU time: Sys=0:00:58.6, User=0:11:41.8
       Memory: 262.4M
       Disk usage: 291.6M
       Time elapsed: 0:20:44.5

**Comment by nicoddemus** [comment]
> Thanks. Perhaps pytest <4 should be added to the recipe? I added it to requirements: run: and test: requires: on my copy.

pytest should appear only on `test/requires` I think:

```yaml
test:
  requires:
    - pytest <4
    - mpmath
```

Also ideally we should add a comment to a related upstream issue, so the pin can be taken out in the future when it scipy is made compatible with pytest 4.

**Comment by tadeu** [comment]
Nice find @FSund! It seems that patch was already released in version 1.1.0: https://github.com/scipy/scipy/commit/ce36940a1367feac90bb4b63a7a52b2c670f2868

This PR is over an older version of SciPy (1.0.1), perhaps rebase into master to see what happens?

**Comment by FSund** [comment]
I tried rebasing into master, but there were a lot of conflicts I an not sure how to resolve.

**Comment by jakirkham** [comment]
Might be easier to merge with `master` so as to resolve conflicts once. For everything outside of the `recipe` directory would prefer the `master` changes. Would resolve the `recipe` changes as you think is best.

**Comment by FSund** [comment]
I tried merging (locally), but it seems like something with the build setup must have changed, because I'm getting the following error which I didn't get before

    C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542800623024\work>for /F "skip=2 tokens=2,*" %A in ('reg query "HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\VisualStudio\14.0" /v InstallDir') do SET "VSINSTALLDIR=%B" 
    ERROR: The system was unable to find the specified registry key or value.

It's looking for the registry key `HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\VisualStudio\14.0\InstallDir`, which was used to determine the Visual Studio install directory on older versions of VS. But it seems like on VS 2017 this registry key is not set anymore [[1](https://social.msdn.microsoft.com/Forums/vstudio/en-US/0a5a7c04-3608-4cd3-ba63-a9d2da6c97ad/no-installdir-value-of-in-vs2017-hklm-registry-key?forum=vssetup)]

I will try setting the registry key manually and look for any further issues during build.

**Comment by FSund** [comment]
I got past the `InstallDir` issue, but here's where progress stopped today

TL;DR:
`distutils.errors.DistutilsExecError: Command "cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt" -IC:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_h_env\Library\include /TcC:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\source.c /FoC:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\source.obj" failed with exit status 127`

Full traceback

    Traceback (most recent call last):
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\distutils\_msvccompiler.py", line 423, in compile
        self.spawn(args)
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\distutils\_msvccompiler.py", line 542, in spawn
        return super().spawn(cmd)
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\ccompiler.py", line 89, in <lambda>
        m = lambda self, *args, **kw: func(self, *args, **kw)
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\ccompiler.py", line 152, in CCompiler_spawn
        raise DistutilsExecError('Command "%s" failed with exit status %d%s' % (cmd, s, msg))
    distutils.errors.DistutilsExecError: Command "cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt" -IC:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_h_env\Library\include /TcC:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\source.c /FoC:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\source.obj" failed with exit status 127

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "setup.py", line 474, in <module>
        setup_package()
      File "setup.py", line 470, in setup_package
        setup(**metadata)
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\core.py", line 135, in setup
        config = configuration()
      File "setup.py", line 388, in configuration
        config.add_subpackage('scipy')
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\misc_util.py", line 1024, in add_subpackage
        caller_level = 2)
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\misc_util.py", line 993, in get_subpackage
        caller_level = caller_level + 1)
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\misc_util.py", line 930, in _get_configuration_from_setup_py
        config = setup_module.configuration(*args)
      File "scipy\setup.py", line 9, in configuration
        config.add_subpackage('cluster')
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\misc_util.py", line 1024, in add_subpackage
        caller_level = 2)
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\misc_util.py", line 993, in get_subpackage
        caller_level = caller_level + 1)
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\misc_util.py", line 930, in _get_configuration_from_setup_py
        config = setup_module.configuration(*args)
      File "scipy\cluster\setup.py", line 16, in configuration
        blas_opt = get_info('lapack_opt')
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\system_info.py", line 432, in get_info
        return cl().get_info(notfound_action)
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\system_info.py", line 624, in get_info
        self.calc_info()
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\system_info.py", line 1534, in calc_info
        openblas_info = get_info('openblas_lapack')
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\system_info.py", line 432, in get_info
        return cl().get_info(notfound_action)
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\system_info.py", line 624, in get_info
        self.calc_info()
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\system_info.py", line 1831, in calc_info
        if not (assume_lapack or self.check_embedded_lapack(info)):
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\site-packages\numpy\distutils\system_info.py", line 1897, in check_embedded_lapack
        obj = c.compile([src], output_dir=tmpdir)
      File "C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_build_env\lib\distutils\_msvccompiler.py", line 425, in compile
        raise CompileError(msg)
    distutils.errors.CompileError: Command "cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt" -IC:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_h_env\Library\include /TcC:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\source.c /FoC:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\source.obj" failed with exit status 127
    source tree in: C:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\work
    Traceback (most recent call last):
      File "C:\Users\IEUser\Miniconda3\envs\build36\Scripts\conda-build-script.py", line 10, in <module>
        sys.exit(main())
      File "C:\Users\IEUser\Miniconda3\envs\build36\lib\site-packages\conda_build\cli\main_build.py", line 439, in main
        execute(sys.argv[1:])
      File "C:\Users\IEUser\Miniconda3\envs\build36\lib\site-packages\conda_build\cli\main_build.py", line 430, in execute
        verify=args.verify, variants=args.variants)
      File "C:\Users\IEUser\Miniconda3\envs\build36\lib\site-packages\conda_build\api.py", line 201, in build
        notest=notest, need_source_download=need_source_download, variants=variants)
      File "C:\Users\IEUser\Miniconda3\envs\build36\lib\site-packages\conda_build\build.py", line 2275, in build_tree
        notest=notest,
      File "C:\Users\IEUser\Miniconda3\envs\build36\lib\site-packages\conda_build\build.py", line 1449, in build
        windows.build(m, build_file, stats=build_stats)
      File "C:\Users\IEUser\Miniconda3\envs\build36\lib\site-packages\conda_build\windows.py", line 316, in build
        check_call_env(cmd, cwd=src_dir, stats=stats)
      File "C:\Users\IEUser\Miniconda3\envs\build36\lib\site-packages\conda_build\utils.py", line 319, in check_call_env
        return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
      File "C:\Users\IEUser\Miniconda3\envs\build36\lib\site-packages\conda_build\utils.py", line 299, in _func_defaulting_env_to_os_environ
        raise subprocess.CalledProcessError(proc.returncode, _args)
    subprocess.CalledProcessError: Command '['cmd.exe', '/c', 'bld.bat']' returned non-zero exit status 1.

**Comment by FSund** [comment]
I did some thinking, tried reinstalling VS Build Tools again, but nothing helped. Then I noticed the a DeprecationWarning just above the wall of text I posted above. Is the build failing on that warning, and not on the `cl.exe` step?

Here's the output from just before the lines I posted in my previous comment

	creating C:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\Users
	creating C:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\Users\IEUser
	creating C:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\Users\IEUser\AppData
	creating C:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\Users\IEUser\AppData\Local
	creating C:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\Users\IEUser\AppData\Local\Temp
	creating C:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8
	cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt" -IC:\Users\IEUser\Miniconda3\envs\build36\conda-bld\scipy_1542805286838\_h_env\Library\include /TcC:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\source.c /FoC:\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\Users\IEUser\AppData\Local\Temp\tmpegu9e4n8\source.obj
	setup.py:114: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
	  import imp

Would the DeprecationWarning trigger a traceback and exiting the build?

**Comment by hmaarrfk** [comment]
I just managed to build numpy 1.11.3 with the windows compilers on python 3.6+

https://github.com/conda-forge/numpy-feedstock/pull/128

Maybe that recipe can be merge and we can try this again?
Should simplify the pinings.

**Comment by ocefpaf** [comment]
@isuruf closing this as stale. I guess that if we are to try this again it is better to start fresh.
